src/examples_localhost/c/event_test.c

```diff
@@ -0,0 +1,56 @@
+#include<event_system.h>
+#include<stdio.h>
+#include<unistd.h>
+
+#define SECOND_TO_NANO(s) s * (uint64_t) 1000000000
+
+const uint64_t heardbeat_interval = SECOND_TO_NANO(1);
+const uint64_t disconnect_interval = SECOND_TO_NANO(5);
+uint64_t last_time;
+
+uint64_t test_get_nanotime() {
+    struct timespec t;
+    clock_gettime(CLOCK_MONOTONIC, &t);
+    return t.tv_sec * 1000000000 + t.tv_nsec;
+}
+
+char send_heardbeat_event(void * carry_data) {
+    uint64_t n_time = test_get_nanotime();
+    printf("time since last call: %8lu us - expected:%8lu us\n", (n_time - last_time) / 1000, heardbeat_interval / 1000);
+    last_time = n_time;
+    return 0;
+}
+
+char disconnect_event(void* carry_data) {
+    printf("disconnecting due to inactivity\n");
+    return 1;
+}
+
+char event_read(void * carry_data) {
+    char buffer[128];
+    ssize_t len = read(STDIN_FILENO, buffer, 127);
+    if (buffer[0] != '\n') {
+        buffer[len] = 0;
+        printf("detected: %s", buffer);
+    }
+    rescedule_event((timed_event*) carry_data);
+    return 0;
+}
+
+int main() {
+    last_time = test_get_nanotime();
+    timed_event t_events[2];
+    t_events[0].callback = send_heardbeat_event;
+    t_events[0].interval = heardbeat_interval;
+    t_events[0].carry_data = NULL;
+
+    t_events[1].callback = disconnect_event;
+    t_events[1].interval = disconnect_interval;
+    t_events[1].carry_data = NULL;
+
+    fd_event f_events[1];
+    f_events[0].callback = event_read;
+    f_events[0].fd = STDIN_FILENO;
+    f_events[0].carry_data = t_events + 1;
+    start_event_loop(t_events, 2, f_events, 1);
+}
```

Comment:
small remark: Adding a final newline has the benefit that adding new lines to a file does not cause git to display a diff on the previous last line (which is unmodified except for the trailing `\n`).